### PR TITLE
Deployment (Beta)

### PR DESCRIPTION
**Pull request by dnc1994**
#### Features

- [x] Dockerization.
- [x] Change `baseUrl` to real domain.
- [x] Get API endpoint from config.
- [x] Use [serve](https://www.npmjs.com/package/serve) for frontend hosting.
- [x] Use Elastic Beanstalk for deployment.

#### Testing
Manual integration testing with deployment on Elastic Beanstalk

#### Reviewer
@yikema

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ import axios from 'axios';
 global.axios = axios;
 // axios.defaults.withCredentials = true;
 
+var apiPrefix = (process.env.NODE_ENV === 'production' ? 'https://api.advex.org' : 'http://localhost:5000');
```
Comment (yikema) [inline src/main.js:9]: what does production here refer to?

File: src/pages/LogOut.vue [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
-     this.$router.push('/auth/login');
-	}
+  created: function() {
+    axios.post(apiPrefix + '/logout',
+      {
+        'user_id': this.$session.get('user_id')
+      },
+      {
+        headers: {
+          'Content-type': 'application/json',
```
Comment (yikema) [inline src/pages/LogOut.vue:13]: Why use this content type after logging out?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM node:9
```
Comment (yikema) [inline Dockerfile:1]: Why is node 9?

File: src/pages/SignUp.vue [hunk 4]
```diff
@@ -47,8 +47,7 @@ export default {
 
   methods: {
     signup: function() {
-      // TODO: get API from config
-      axios.post('http://localhost:5000/users',
+      axios.post(apiPrefix + '/users',
         {
           'nickname': this.nickname,
```
Comment (yikema) [inline src/pages/SignUp.vue:52]: Do we need the password of the user here?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+ADD . /fe
+WORKDIR /fe
+
+RUN npm install
+RUN npm run build
+RUN npm install serve --g
+
+EXPOSE 4000
```
Comment (yikema) [inline Dockerfile:10]: Is 4000 the port?

**Review by yikema — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (dnc1994) [reply Dockerfile:1]: 9 is the current node version.

File: Dockerfile (hunk 5 above)
Comment (dnc1994) [reply Dockerfile:10]: Yes.

File: src/pages/SignUp.vue (hunk 4 above)
Comment (dnc1994) [reply src/pages/SignUp.vue:52]: Yes.

File: src/pages/LogOut.vue (hunk 2 above)
Comment (dnc1994) [reply src/pages/LogOut.vue:13]: JSON is used in all communications with BE.

File: src/main.js (hunk 1 above)
Comment (dnc1994) [reply src/main.js:9]: It means production environment. When you use `npm run build` to generate the minified files that will be used in production, this `NODE_ENV` will be set to `production`. We use it to distinguish between local debugging mode and production mode.

**Pull request merged**